src/Microsoft.Identity.Web/DownstreamWebApiSupport/MicrosoftIdentityAuthenticationMessageHandlerHttpClientBuilderExtensions.cs

```diff
@@ -18,19 +18,19 @@ public static class MicrosoftIdentityAuthenticationMessageHandlerHttpClientBuild
         /// Adds a named Microsoft Identity user authentication message handler related to a specific configuration section.
         /// </summary>
         /// <param name="builder">Builder.</param>
-        /// <param name="serviceName">Name of the configuration for the service.</param>
         /// <param name="configuration">Configuration.</param>
         /// <returns>The builder for chaining.</returns>
         public static IHttpClientBuilder AddMicrosoftIdentityUserAuthenticationHandler(

```

Comment:
Given it would be a breaking change, I think a better fix would be to:
- keep this method (maybe with an obsolete attribute)
- add an additional override that calls this one passing `builder.Name`?

```CSharp
 /// Adds a named Microsoft Identity user authentication message handler related to a specific configuration section.
        /// </summary>
        /// <param name="builder">Builder.</param>
        /// <param name="serviceName">Name of the configuration for the service.</param>
        /// <param name="configuration">Configuration.</param>
        /// <returns>The builder for chaining.</returns>
        public static IHttpClientBuilder AddMicrosoftIdentityUserAuthenticationHandler(
 

 
Given it would be a breaking change, I think a better fix would be to:
- keep this method (maybe with an obsolete attribute)
- add an additional override that calls this one passing `builder.Name`?

```CSharp
```
 public static IHttpClientBuilder AddMicrosoftIdentityUserAuthenticationHandler(
            this IHttpClientBuilder builder,
            IConfiguration configuration)
    {
      if (builder == null)
            {
                throw new ArgumentNullException(nameof(builder));
            }
     builder.AddMicrosoftIdentityUserAuthenticationHandler(builder.Name, configuration);
    }
```

Reply:
Sounds good! I pushed a new change that:

1. Creates new overloads that don't accept `serviceName` and uses the HttpClient name instead
2. Marks the old overloads as obsolete
3. Updated PR title and description to match the new behavior.

Please let me know what you think!

I think I've addressed the concerns in this conversation, so I'm marking it as resolved. If you have questions or concerns, please reactivate. Thanks!